modules/entrezdirect/esummary/main.nf

```diff
@@ -0,0 +1,52 @@
+process ENTREZDIRECT_ESUMMARY {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::entrez-direct=16.2" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/entrez-direct:16.2--he881be0_1':
+        'quay.io/biocontainers/entrez-direct:16.2--he881be0_1' }"
+
+    input:
+    tuple val(meta), val(uid), path(uids_file)
+    val database
+
+    output:
+    tuple val(meta), path("*.xml"), emit: xml_esummary
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def input_id = uid ? "-id ${uid}" : ''
+    def input_file = uids_file.name != 'NO_FILE' ? "-input $uids_file" : ''
```

Comment:
Is `NO_FILE` an internal NXF thing? I've not seen that before? That could be useful if so...

Reply:
No, you only need an string to specify the absence of the file:  [nextflow-io/patterns/docs/optional-input](https://github.com/nextflow-io/patterns/blob/master/docs/optional-input.adoc), here is the full documentation [Nextflow Patterns: A curated collection of Nextflow implementation patterns](https://github.com/nextflow-io/patterns).